lib/gc/Transforms/TilingUsingInterfaceX.cpp

```diff
@@ -681,6 +681,91 @@ fixTerminatorSCFInParallel(RewriterBase &rewriter, scf::ForallOp newForallOp,
   }
 }
 
+/// To solve following residual topology:
+///
+///   op1
+///   /  \
+///   |  op2
+///   |   /
+///   op3
+///
+/// Where the possible IR may appears like below before fusing `op3`:
+///
+/// ```
+/// %1:2 = scf.for() {
+///     %2 = tiled_op1
+///     %3 = tiled_op2
+///     %4 = insert_slice %2
+///     %5 = insert_slice %3
+///     yield %4, %5
+/// }
+/// op3 ins(%1#1, %1#2)
+/// ```
+///
+/// Although current fuse consumer only accepts single one candidate slice such
+/// as `%4`, in fact, `%5` is another regarding slice which produces the operand
+/// of untiled `op3`. Thus, is is necessary to find all of them and ensure that
+/// they are pretty matched.
+///
+/// @param loopOp: loop operation
+/// @param consumerOp: consumer operation
+/// @param curOperandNumber: operand number of current candidate slice
+/// @param otherOperandSliceMap: operand number to other slice mapping
+/// @return LogicalResult: Return `success` only if all operand slice are found.
+static LogicalResult findOtherOperandSlice(
+    Operation *loopOp, Operation *consumerOp, unsigned curOperandNumber,
+    DenseMap<unsigned, OffsetSizeAndStrideOpInterface> &otherOperandSliceMap) {
+  otherOperandSliceMap.clear();
+  SmallVector<OpOperand *> otherUseOfLoop;
+  for (auto &&[en, v] : llvm::enumerate(consumerOp->getOpOperands())) {
+    if (en == curOperandNumber)
+      continue;
+    if (v.get().getDefiningOp() == loopOp) {
+      otherUseOfLoop.push_back(&v);
+    }
+  }
+  if (otherUseOfLoop.empty())
+    return success();
+  if (auto forallOp = dyn_cast<scf::ForallOp>(loopOp)) {
+    for (auto &use : otherUseOfLoop) {
+      tensor::ParallelInsertSliceOp sliceOp;
+      for (auto &useOfResult :
+           forallOp
+               .getRegionIterArgs()[dyn_cast<OpResult>(use->get())
+                                        .getResultNumber()]
+               .getUses()) {
+        if (isa<tensor::ParallelInsertSliceOp>(useOfResult.getOwner())) {
+          if (llvm::detail::isPresent(sliceOp))
+            return failure();
+          sliceOp = cast<tensor::ParallelInsertSliceOp>(useOfResult.getOwner());
+          otherOperandSliceMap[use->getOperandNumber()] =
+              cast<OffsetSizeAndStrideOpInterface>(sliceOp.getOperation());
+        }
+      }
+      if (!llvm::detail::isPresent(sliceOp))
+        return failure();
+    }
+  } else if (auto forOp = dyn_cast<scf::ForOp>(loopOp)) {
+    for (auto &use : otherUseOfLoop) {
+      OpOperand *yieldValue = forOp.getTiedLoopYieldedValue(
+          forOp.getTiedLoopRegionIterArg(dyn_cast<OpResult>(use->get())));
+      Operation *yieldValueProducer = yieldValue->get().getDefiningOp();
+      while (!isa<tensor::InsertSliceOp>(yieldValueProducer)) {
+        if (!isa<scf::ForOp>(yieldValueProducer))
```

Comment:
I know your concern, but considering the context of this function, actually there only exist nest `scf::ForOp` s instead of any other combination as you said. This condition could be ensured by dispatchment inside`tileAndFuseConsumerOfSlice`.

Reply:
Got it. Thanks for the explanation!